rust/src/applayer.rs

```diff
@@ -383,6 +386,17 @@ macro_rules! build_slice {
     ($buf:ident, $len:expr) => ( std::slice::from_raw_parts($buf, $len) );
 }
 
+#[macro_export]
+macro_rules! build_slice_null {
+    ($buf:ident, $len:expr) => {
+        if $buf.is_null() && $len == 0 {
+            &[]
+        } else {
+            std::slice::from_raw_parts($buf, $len)
+        }
+    };
+}
```

Comment:
Ignore this comment for now I think.  Once fixed, I want to do some other cleanup like hiding unsafe inside a macro without doing anything to make it safe :)

Reply:
Changing in next version, to have something for this specific HTTP2 use case

This is not a function because rustc does not like it :
```
error[E0106]: missing lifetime specifier
    --> src/http2/detect.rs:1006:60
     |
1006 | fn build_slice_null(input: *const u8, input_len: usize) -> &[u8] {
     |                                                            ^ expected named lifetime parameter
     |
     = help: this function's return type contains a borrowed value, but there is no value for it to be borrowed from
help: consider using the `'static` lifetime, but this is uncommon unless you're returning a borrowed value from a `const` or a `static`
```